dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo

```diff
@@ -0,0 +1,105 @@
+    Placement(visible = true, transformation(origin = {-120, 20}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+
+  Modelica.Blocks.Interfaces.RealOutput URefPu(start = URef0 / UNom) "Voltage reference for the regulation in p.u (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanOutput blocked "Wheter the static var compensator is blocked due to very low voltages" annotation(
+    Placement(visible = true, transformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  ModeConnector mode(value(start = Mode0)) "Current mode of the static var compensator" annotation(
+    Placement(visible = true, transformation(origin = {100, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {110, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  ModeConnector mode_auto(value(start = Mode0)) "Mode of the static var compensator when in automatic configuration";
+  ModeConnector mode_manual(value(start = Mode0)) "Mode of the static var compensator when in manual configuration";
+
+  Modelica.Blocks.Interfaces.IntegerInput setMode "Mode selected when in manual configuration" annotation(
+    Placement(visible = true, transformation(origin = {-120, -20}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanInput selectModeAuto "Wheter the static var compensator is in automatic configuration" annotation(
+    Placement(visible = true, transformation(origin = {-120, -60}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 28}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+
+  Types.Time timerModeChangeUp(start = Modelica.Constants.inf) "Timer for the transition from standby to running mode for high voltage values";
+  Types.Time timerModeChangeDown(start = Modelica.Constants.inf) "Timer for the transition from standby to running mode for low voltage values";
+  Types.VoltageModule URef_auto(start = URef0) "Voltage reference for the regulation if the static var compensator switches from standy to running mode automatically in kV";
+
+protected
+  parameter Types.VoltageModule URef0  "Start value of voltage reference in kV";
+  parameter Mode Mode0 "Start value for mode";
+
+equation
+  // Timer for the transition from standby to running mode for high voltage values
+  when UPu*UNom > UThresholdUp then
+    timerModeChangeUp = time;
+  elsewhen UPu*UNom < UThresholdUp then
+    timerModeChangeUp = Modelica.Constants.inf;
+  end when;
```

Comment:
We can easily avoid a few operations there without loosing in clarity I think by dividing the threshold by UNom (doing a final parameter) and always compare in p.u.
-> it avoids all the multiplication (the optimisation is perhaps done by OM but I'm not sure: too be double checked with @rosiereflo )

Reply:
fixed